src/firebaseConfig.ts

```diff
@@ -5,35 +5,36 @@
 // 'npm run generate:json-schema' to regenerate the schema files.
 //
 
-export type DatabaseConfig =
+// base configs
+type DeployAsset = {
+  predeploy?: string | string[];
+  postdeploy?: string | string[];
+};
```

Comment:
The documentation for pre/post hooks https://firebase.google.com/docs/cli#hooks only states
>You can set up predeploy and postdeploy hooks for any of the [assets that you can deploy](https://firebase.google.com/docs/cli#deployment)

Reply:
That sounds reasonable to me! Nit: maybe `Deployable` is a better name?